### ISSUE TITLE
Candy Cafe: Cannot swap candies

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#10526]
### Describe the bug

Link to SWF file: https://www.spiele-umsonst.de/azad/downloads/flashgames/candycafe2.swf

In [Candy Cafe](https://www.izzygames.com/candy-cafe-t5788.html), Ruffle softlocks on this screen:
![image](https://user-images.githubusercontent.com/71368227/229381284-838230e7-6511-4e5e-9d33-801dfadff516.png)

### Expected behavior

It should be possible to click the candies to swap them.

### Affected platform

Desktop app

### Operating system

Windows 10

**Comment by Lord-McSweeney** [comment]
On web, an XML error shows up, but no error appears on the desktop version. It's probable that the error is unrelated to the problem.

**Comment by n0samu** [comment]
I'm seeing 2 errors:
```
ERROR run_frame:run_all_phases_avm2: ruffle_core::display_object::movie_clip: Error occured when running AVM2 frame script: ReferenceError: Error #1069: Property objectEncoding not found on flash.net.SharedObject and there is no default value.
        at save_fla::MainTimeline/frame1()
```
```
ERROR run_frame:run_all_phases_avm2: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "complete", class: flash.events::Event, ptr: 0x271ce3ab1f8 }) to handler FunctionObject(FunctionObject { ptr: 0x271e5ba3fe8 }) : TypeError: Error #1009: Cannot access a property or method of a null object reference. (accessing field: data)
        at save_fla::MainTimeline/getLogPair()
        at IDNET/onCookieLoad()
```

**Comment by desertking** [comment]
Error still happening - can't swipe the candies

**Comment by desertking** [comment]
something with the last updates patched it :)

**Comment by ArbestRiAgain** [comment]
Close this issue

**Comment by desertking** [comment]
I can just wondered if anyone wanted to add the code number which caused it to work :D

**Comment by n0samu** [comment]
Fixed by #12553
